cmd/livepeer/livepeer.go

```diff
@@ -868,6 +868,21 @@ func setupOrchestrator(ctx context.Context, n *core.LivepeerNode, initializeRoun
 		return err
 	}
 
+	// add orchestrator to DB
+	orch, err := n.Eth.GetTranscoder(n.Eth.Account().Address)
+	if err != nil {
+		return err
+	}
+
+	err = n.Database.UpdateOrch(&common.DBOrch{
+		EthereumAddr:      n.Eth.Account().Address.Hex(),
+		ActivationRound:   orch.ActivationRound.Int64(),
+		DeactivationRound: orch.DeactivationRound.Int64(),
```

Comment:
While true, if deactivation round starts overflowing it will break any form of doing an active check anyway and we'll have to start storing these values as `BLOB` and converting them to `big.Int` when retrieving them from the DB. 

Added it for sanity in 893bd1d

Also added unit testing for setupOrchestrator 317ba32

Reply:
> While true, if deactivation round starts overflowing it will break any form of doing an active check anyway

`DeactivationRound.Int64()` for an active orchestrator would always overflow when stored in an int64 since the value stored on-chain is a max uint256. In this case, we would end up storing a large negative integer in the `deactivationRound` column of the DB. As a result, every active orchestrator tracked via the DB would be considered inactive in the current round because the node thinks that the orchestrator's deactivation round is in the past. Using the `common.ToInt64()` helper should ensure that we never end up in such a situation.

Reply:
I don't think we're talking about the same thing here. 

That alleviates the problem currently, but whenever the rounds on mainnet become larger than maxInt64 ,the current way of doing this will break anyway, regardless if we deal with overflows or not at this stage. Which makes dealing with overflows currently in this scenario kind of trivial. 

It won't occur currently, and when it starts occuring we'll need to change the codebase anyway to use *big.Int across the board and BLOB for storage of these integers. 

Example: 
1. Orchestrator became active in round `math.MaxInt64`
2. O's deactivation round gets set to `math.MaxUint256` in the contracts, but `math.MaxInt64` on the client
3. Orchestrator shows as inactive because `DeactivationRound == ActivationRound`

Reply:
Having a round overflowing MaxInt64 is 5e16 years away... what vector other than a protocol bug would lead us to receiving a value in that neighborhood? Are there other consequences more damaging than marking an orchestrator as inactive? (Which seems relatively minor, all things considered.)

I am in favor of avoiding BLOBs and bignum arithmetic wherever possible.

Reply:
@j0sh Don't get me wrong, I agree with you here. 

> what vector other than a protocol bug would lead us to receiving a value in that neighborhood?

None, but that exact same would true for the added `ToInt64` helper here, wouldn't it?  Which is why it's more for sanity than absolute necessity. That's all I was trying to say.

Reply:
> Which is why it's more for sanity than absolute necessity. 

I'm assuming that you're just talking about using `common.ToInt64()`  for `ActivationRound` in the above comment?

It is true that using `common.ToInt64()` for `ActivationRound` would just be for consistency because its value is unlikely to ever overflow an `int64` given the current round length on mainnet.

However, assuming that we want to avoid using `big.Int`/`BLOB` here, using `common.Int64()` *is* an absolute necessity for `DeactivationRound` because its value will overflow an `int64` for every active orchestrator since the value on-chain is a max uint256.

Reply:
In the `OrchestratorWatcher` we use `const maxFutureRound` whenever a transcoder becomes activated to mimic the on-chain behaviour: https://github.com/livepeer/go-livepeer/blob/f41ad9982d6254f1cf217c6834b15c0a02451521/eth/watchers/orchestratorwatcher.go#L16

But I do concur that the helper  (`ToInt64()`) comes in handy dealing with this issue in `ethOrchToDBOrch` and `setupOrchestrator`.